forge-gui/res/cardsfolder/upcoming/teferi_temporal_pilgrim.txt

```diff
@@ -0,0 +1,13 @@
+Name:Teferi, Temporal Pilgrim
+ManaCost:3 U U
+Types:Legendary Planeswalker Teferi
+Loyalty:4
+T:Mode$ Drawn | ValidCard$ Card.YouCtrl | TriggerZones$ Battlefield | Execute$ TrigPutCounter | TriggerDescription$ Whenever you draw a card, put a loyalty counter on CARDNAME.
+SVar:TrigPutCounter:DB$ PutCounter | Defined$ Self | CounterType$ LOYALTY | CounterNum$ 1
+A:AB$ Draw | Cost$ AddCounter<0/LOYALTY> | NumCards$ 1 | Planeswalker$ True | SpellDescription$ Draw a card.
+A:AB$ Token | Cost$ SubCounter<2/LOYALTY> | Planeswalker$ True | TokenScript$ u_2_2_spirit_vigilance_drawcounter | TokenAmount$ 1 | TokenOwner$ You | SpellDescription$ Create a 2/2 blue Spirit creature token with vigilance and "Whenever you draw a card, put a +1/+1 counter on this creature."
+A:AB$ ChooseCard | Cost$ SubCounter<12/LOYALTY> | Planeswalker$ True | Ultimate$ True | ValidTgts$ Opponent | TgtPrompt$ Select target opponent | Mandatory$ True | Choices$ Permanent.TargetedPlayerCtrl | SubAbility$ DBChangeZone | RememberChosen$ True | SpellDescription$ Target opponent chooses a permanent they control and returns it to its owner's hand. Then they shuffle each nonland permanent they control into its owner's library.
```

Comment:
Gonna need cleanup, also just work with ChosenCard

Reply:
You're right, switched the solution to ChosenCard to curb any memory issues with remembered. Thanks!